Classes/Emogrifier.php

```diff
@@ -530,6 +530,13 @@ private function fillStyleAttributesWithMergedStyles()
     private function generateStyleStringFromDeclarationsArrays(array $oldStyles, array $newStyles)
     {
         $combinedStyles = array_merge($oldStyles, $newStyles);
+
+        foreach ($oldStyles as $attribute => $expression) {
+            if (isset($newStyles[$attribute]) && strtolower(substr($expression, -10)) === '!important') {
```

Comment:
Do we need to use substr here, or would it suffice to use stripos to check whether !important is part of the string?